portalnet/src/utp_controller.rs

```diff
@@ -15,33 +25,253 @@ use crate::discovery::UtpEnr;
 pub struct UtpController {
     pub inbound_utp_transfer_semaphore: Arc<Semaphore>,
     pub outbound_utp_transfer_semaphore: Arc<Semaphore>,
-    /// uTP socket.
     utp_socket: Arc<UtpSocket<UtpEnr>>,
+    metrics: OverlayMetricsReporter,
+}
+
+lazy_static! {
+    /// The default configuration to use for uTP connections.
+    pub static ref UTP_CONN_CFG: ConnectionConfig = ConnectionConfig { max_packet_size: 1024, ..Default::default()};
 }
 
 impl UtpController {
-    pub fn new(utp_transfer_limit: usize, utp_socket: Arc<UtpSocket<UtpEnr>>) -> Self {
+    pub fn new(
+        utp_transfer_limit: usize,
+        utp_socket: Arc<UtpSocket<UtpEnr>>,
+        metrics: OverlayMetricsReporter,
+    ) -> Self {
         Self {
             utp_socket,
             inbound_utp_transfer_semaphore: Arc::new(Semaphore::new(utp_transfer_limit)),
             outbound_utp_transfer_semaphore: Arc::new(Semaphore::new(utp_transfer_limit)),
+            metrics,
+        }
+    }
+
+    /// Connect with a peer given the connection id, and return the data received from the peer.
+    pub async fn connect_inbound_stream(
```

Comment:
Can't the logic in `connect_inbound_stream` and `accept_inbound_stream` be condensed? there is a ton of duplicate code

Reply:
I don't think there's a ton of duplicate code. But I did extract the reading of the utp stream data into `receive_utp_content`, though I'm not too sure we won too many lines with that change